eve/io/mongo/mongo.py

```diff
@@ -66,7 +66,8 @@ class Mongo(DataLayer):
         'datetime': str_to_date,
         'integer': lambda value: int(value) if value is not None else None,
         'float': lambda value: float(value) if value is not None else None,
-        'number': lambda val: json.loads(val) if val is not None else None
+        'number': lambda val: json.loads(val) if isinstance(val, basestring)
```

Comment:
This will fail under Python 3.x. Try importing and using `base_str`  from `utils` ([link](https://github.com/nicolaiarocci/eve/blob/develop/eve/utils.py#L436)).